File: unity/oelMVCS/Assets/XTC/oelMVCS/Scripts/Model/ModelCenter.cs
using System.Collections.Generic;

namespace XTC.oelMVCS
{
    public class ModelCenter
    {
        // 数据列表
        private Dictionary<string, Model.Inner> models = new Dictionary<string, Model.Inner>();
        // 状态列表
        private Dictionary<string, Model.Status> status = new Dictionary<string, Model.Status>();

        private Board board_
        {
            get;
            set;
        }

        public ModelCenter(Board _board)
        {
            board_ = _board;
        }

        public Error Register(string _uuid, Model.Inner _inner)
        {
            board_.logger.Info("register model {0}", _uuid);

            if (models.ContainsKey(_uuid))
                return Error.NewAccessErr("model {0} exists", _uuid);
            models[_uuid] = _inner;
            return Error.OK;
        }

        public Error Cancel(string _uuid)
        {
            board_.logger.Info("cancel model {0}", _uuid);

            if (!models.ContainsKey(_uuid))
                return Error.NewAccessErr("model {0} not found", _uuid);
            models.Remove(_uuid);
            return Error.OK;
        }

        public Model.Inner FindModel(string _uuid)
        {
            if (models.ContainsKey(_uuid))
                return models[_uuid];
            return null;
        }

        public void Setup()
        {
            foreach (Model.Inner inner in models.Values)
            {
                inner.Setup(board_);
            }
        }

        public void Dismantle()
        {
            foreach (Model.Inner inner in models.Values)
            {
                inner.Dismantle();
            }
        }

        public void Broadcast(string _action, Model.Status _status, object _data)
        {
            board_.viewCenter.HandleAction(_action, _status, _data);
        }
    }
}//namespace

File: python/oelMVCS/ModelCenter.py
# -*- coding: utf-8 -*-

class StatusCenter:
    def __init__(self, _framework):
        self.__framework = _framework
        self.status = {}

    def Register(self, _uuid, _status):
        self.__framework.logger.Info('register status %s' % (_uuid))
        if _uuid in self.status:
            raise KeyError('status is exists')
        self.status[_uuid] = _status

    def Cancel(self, _uuid):
        self.__framework.logger.Info('cancel status %s' % (_uuid))
        if not _uuid in self.status:
            raise KeyError('status not found')
        del self.status[_uuid]

    def Find(self, _uuid):
        return self.status[_uuid]

class ModelCenter:

    def __init__(self, _framework):
        self.__framework = _framework
        self.models = {}

    def Register(self, _uuid, _model):
        self.__framework.logger.Info('register model %s' % (_uuid))
        if _uuid in self.models:
            raise KeyError('model is exists')
        self.models[_uuid] = _model

    def Cancel(self, _uuid):
        self.__framework.logger.Info('cancel model %s' % (_uuid))
        if not _uuid in self.models:
            raise KeyError('model not found')
        del self.models[_uuid]

    def Find(self, _uuid):
        return self.models[_uuid]

    def Setup(self):
        for v in self.models.values():
            v.Setup(self.__framework)

    def Dismantle(self):
        for v in self.models.values():
            v.Dismantle()

    def broadcast(self, _action, _status, _data):
        self.__framework.viewCenter.handleAction(_action, _status, _data)
